src/components/shared/Validations.tsx

```diff
@@ -1,90 +1,161 @@
-import { safeStringify } from '@stoplight/json';
-import { Dictionary } from '@stoplight/types';
-import { Popover } from '@stoplight/ui-kit';
+import { Dictionary, Primitive } from '@stoplight/types';
+import { Box, Flex, Text } from '@stoplight/mosaic';
+import { capitalize, keys, omit, pick, pickBy, uniq } from 'lodash';
 import * as React from 'react';
-
-import { ViewModeContext } from '../../contexts';
+import { RegularNode } from '@stoplight/json-schema-tree';
 
 export interface IValidations {
-  required: boolean;
-  deprecated: boolean;
   validations: Dictionary<unknown>;
 }
 
+const numberValidationNames = [
+  'minimum',
+  'maximum',
+  'minLength',
+  'maxLength',
+  'minItems',
+  'maxItems',
+  'exclusiveMinimum',
+  'exclusiveMaximum',
+];
+
+const excludedValidations = [
+  'exclusiveMinimum',
+  'exclusiveMaximum',
+  'readOnly',
+  'writeOnly'
+];
+
 export const Validations: React.FunctionComponent<IValidations> = ({
-  required,
-  deprecated,
-  validations: { readOnly, writeOnly, ...validations },
+  validations,
 }) => {
-  const viewMode = React.useContext(ViewModeContext);
-  const validationCount = Object.keys(validations).length;
-
-  const requiredElem = (
-    <div>
-      {required && <span className="ml-2 text-orange-7 dark:text-orange-6">required</span>}
-      {validationCount ? <span className="ml-1 text-darken-7 dark:text-lighten-6">+{validationCount}</span> : null}
-    </div>
+  const numberValidations = pick(validations, numberValidationNames);
+  const booleanValidations = omit(
+    pickBy(validations, v => ['true', 'false'].includes(String(v))),
+    excludedValidations,
   );
```

Comment:
Why do you do the exclusion out here for booleans, but not for numbers? I prefer the way you do it for booleans here, so the component doesn't have to worry about what to show, but either way is fine.

Reply:
Booleans are simpler case, for number validations we would need to exclude different more names, so I assumed listing number ones will be easier.